contracts/settling_game/utils/game_structs.cairo

```diff
@@ -13,75 +20,143 @@ struct RealmData:
+    member Farms : felt
+    member Hamlet : felt
 end
 
 struct RealmBuildingCostIds:
-    member resource_1 : felt  
-    member resource_2 : felt  
-    member resource_3 : felt   
-    member resource_4 : felt  
+    member resource_1 : felt
+    member resource_2 : felt
+    member resource_3 : felt
+    member resource_4 : felt
     member resource_5 : felt
-    member resource_6 : felt  
-    member resource_7 : felt  
-    member resource_8 : felt   
-    member resource_9 : felt  
-    member resource_10 : felt           
+    member resource_6 : felt
+    member resource_7 : felt
+    member resource_8 : felt
+    member resource_9 : felt
+    member resource_10 : felt
+end
+
+struct RealmBuildingCostValues:
+    member resource_1_values : felt
+    member resource_2_values : felt
+    member resource_3_values : felt
+    member resource_4_values : felt
+    member resource_5_values : felt
+    member resource_6_values : felt
+    member resource_7_values : felt
+    member resource_8_values : felt
+    member resource_9_values : felt
+    member resource_10_values : felt
 end
 
-struct RealmBuildingCostValues:     
-    member resource_1_values : felt  
-    member resource_2_values  : felt  
-    member resource_3_values  : felt   
-    member resource_4_values  : felt  
-    member resource_5_values  : felt       
-    member resource_6_values : felt  
-    member resource_7_values  : felt  
-    member resource_8_values  : felt   
-    member resource_9_values  : felt  
-    member resource_10_values  : felt     
-end
\ No newline at end of file
+namespace RealmBuildingsIds:
+    const Fairgrounds = 1
+    const RoyalReserve = 2
+    const GrandMarket = 3
+    const Castle = 4
+    const Guild = 5
+    const OfficerAcademy = 6
+    const Granary = 7
+    const Housing = 8
+    const Amphitheater = 9
+    const Carpenter = 10
+    const School = 11
+    const Symposium = 12
+    const LogisticsOffice = 13
+    const ExplorersGuild = 14
+    const ParadeGrounds = 15
+    const ResourceFacility = 16
+    const Dock = 17
+    const Fishmonger = 18
+    const Farms = 19
+    const Hamlet = 20
+end
+
+namespace RealmBuildingLimitTraitsIds:
+    const Fairgrounds = TraitsIds.Region
+    const RoyalReserve = TraitsIds.Region
+    const GrandMarket = TraitsIds.Region
+    const Castle = TraitsIds.Region
+    const Guild = TraitsIds.Region
+    const OfficerAcademy = TraitsIds.Region
+    const Granary = TraitsIds.City
+    const Housing = TraitsIds.City
+    const Amphitheater = TraitsIds.City
+    const Carpenter = TraitsIds.City
+    const School = TraitsIds.City
+    const Symposium = TraitsIds.City
+    const LogisticsOffice = TraitsIds.City
+    const ExplorersGuild = TraitsIds.City
+    const ParadeGrounds = TraitsIds.City
+    const ResourceFacility = TraitsIds.City
+    const Dock = TraitsIds.Harbour
+    const Fishmonger = TraitsIds.Harbour
+    const Farms = TraitsIds.River
+    const Hamlet = TraitsIds.River
+end
+
```

Comment:
Are nested namespaces possible?:
```
namespace RealmBuildingData:
    namespace Fairgrounds:
         const Id = 1
         const TraitLimit = TraitsIds.Region
```
Just to have 1 source of truth.

Reply:
Not sure if it's possible, but I like this idea a lot.